php/class-media.php

```diff
@@ -365,6 +366,63 @@ public function is_uploadable_media( $attachment_id ) {
 		return apply_filters( 'cloudinary_is_uploadable_media', $is_uploadable, $media_host );
 	}
 
+	/**
+	 * Is the media item size allowed to be uploaded.
+	 * Checks against the account limits.
+	 *
+	 * @param int $attachment_id The attachment ID.
+	 *
+	 * @return bool
+	 */
+	public function is_oversize_media( $attachment_id ) {
+		static $is_oversize = array();
+
+		if ( isset( $is_oversize[ $attachment_id ] ) ) {
+			return $is_oversize[ $attachment_id ];
+		}
+
+		$is_oversize[ $attachment_id ] = false;
+		$file_size                     = $this->get_attachment_file_size( $attachment_id );
+		$max_size                      = ( wp_attachment_is_image( $attachment_id ) ? 'image_max_size_bytes' : 'video_max_size_bytes' );
+		$limit                         = $this->plugin->components['connect']->usage['media_limits'][ $max_size ];
+
+		if ( $file_size > $limit ) {
+			$is_oversize[ $attachment_id ] = true;
```

Comment:
 ​We can simplify this with `$​is_oversize​[ ​$​attachment_id​ ]​ = ​$file_size > $limit​;`